sdks/rust/src/internals/serialize.rs

```diff
@@ -0,0 +1,124 @@
+use std::collections::HashMap;
+use std::iter::Iterator;
+use std::marker::PhantomData;
+
+use std::any::Any;
+use std::boxed::Box;
+use std::sync::Mutex;
+
+use once_cell::sync::Lazy;
+
+static SERIALIZED_FNS: Lazy<Mutex<HashMap<String, Box<dyn Any + Sync + Send>>>> =
+    Lazy::new(|| Mutex::new(HashMap::new()));
+
+pub fn serialize_fn<T: Any + Sync + Send>(obj: Box<T>) -> String {
+    let name = format!("object{}", SERIALIZED_FNS.lock().unwrap().len());
+    SERIALIZED_FNS.lock().unwrap().insert(name.to_string(), obj);
```

Comment:
As mentioned, there's the possibility that two threads store the same name and overwrite one another's entry. Using a single mutex guard should do the trick.
```suggestion
    let mut binding = SERIALIZED_FNS.lock().unwrap();
    let name = format!("object{}", binding.len());
    binding.insert(name.to_string(), obj);
```